examples/angular/auto-refetching/src/app/services/tasks.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpClient } from '@angular/common/http'
+import { Injectable, inject } from '@angular/core'
+import {
+  QueryClient,
+  mutationOptions,
+} from '@tanstack/angular-query-experimental'
+import { Observable, lastValueFrom, map, tap } from 'rxjs'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TasksService {
+  #queryClient = inject(QueryClient) // Manages query state and caching
+  #http = inject(HttpClient) // Handles HTTP requests
+
+  /**
+   * Fetches all tasks from the API.
+   * Returns an observable containing an array of task strings.
+   */
+  allTasks$ = () => this.#http.get<Array<string>>('/api/tasks')
```

Comment:
For consistency with `mutationOptions` in this same file it would be nice to use `queryOptions` here